### ISSUE TITLE
Elixir code output needs to be formatted and some atoms need double quotes

### ISSUE THREAD

**Issue by eksperimental** [esl/gradient#30]
Hi @erszcz, we talked over the Elixir Forum,
I will be reporting some of the stuff that I find difficult to deal with Gradient

This is my project and commit
https://github.com/eksperimental/beam_langs_meta_data/tree/26ecd958475bedd5c0e4204814e7437484acdff4

`mix gradient` give me this:

`lib/beam_langs_meta_data.ex: The map on line 242 is expected to have type %{required(:latest) => version_string(), required(:releases) => nonempty_keyword(atom(), elixir_release_data())} but it has type %{required(:latest) => binary(), required(:releases) => nonempty_list({:1.13.0, %{optional(:assets) => nonempty_list(%{optional(:content_type) => binary(), optional(:created_at) => binary(), optional(:download_url) => binary(), optional(:id) => 50943932, optional(:label) => nil, optional(:name) => binary(), optional(:node_id) => binary(), optional(:size) => 2023670, optional(:state) => binary(), optional(:url) => binary()} | %{optional(:content_type) => binary(), optional(:created_at) => binary(), optional(:download_url) => binary(), optional(:id) => 50943934, optional(:label) => nil, optional(:name) => binary(), optional(:node_id) => binary(), optional(:size) => 6230020, optional(:state) => binary(), optional(:url) => binary()}), optional(:assets_url) => binary(), optional(:body) => binary(), optional(:created_at) => binary(), optional(:draft) => false, optional(:id) => 54599470, optional(:name) => binary(), optional(:node_id) => binary(), optional(:prerelease) => false, optional(:published_at) => binary(), optional(:release_url) => binary(), optional(:tag_name) => binary(), optional(:tarball_url) => binary(), optional(:target_commitish) => binary(), optional(:upload_url) => binary(), optional(:url) => binary(), optional(:zipball_url) => binary()}} | {:1.13.0-rc.0, %{optional(:assets) => nonempty_list(%{optional(:content_type) => binary(), optional(:created_at) => binary(), optional(:download_url) => binary(), optional(:id) => 48381634, optional(:label) => nil, optional(:name) => binary(), optional(:node_id) => binary(), optional(:size) => 2020048, optional(:state) => binary(), optional(:url) => binary()} | %{optional(:content_type) => binary(), optional(:created_at) => binary(), optional(:download_url) => binary(), optional(:id) => 48381646, optional(:label) => nil, optional(:name) => binary(), optional(:node_id) => binary(), optional(:size) => 6223789, optional(:state) => binary(), optional(:url) => binary()}), optional(:assets_url) => binary(), optional(:body) => binary(), optional(:created_at) => binary(), optional(:draft) => false, optional(:id) => 52477144, optional(:name) => binary(), optional(:node_id) => binary(), optional(:prerelease) => true, optional(:published_at) => binary(), optional(:release_url) => binary(), optional(:tag_name) => binary(), optional(:tarball_url) => binary(), optional(:target_commitish) => binary(), optional(:upload_url) => binary(), optional(:url) => binary(), optional(:zipball_url) => binary()}} | {:1.13.0-rc.1, %{optional(:assets) => nonempty_list(%{optional(:content_type) => binary(), optional(:created_at) => binary(), optional(:download_url) => binary(), optional(:id) => 49611887, optional(:label) => nil, optional(:name) => binary(), optional(:node_id) => binary(), optional(:size) => 2016743, optional(:state) => binary(), optional(:url) => binary()} | %{optional(:content_type) => binary(), optional(:created_at) => binary(), optional(:download_url) => binary(), optional(:id) => 49611889, optional(:label) => nil, optional(:name) => binary(), optional(:node_id) => binary(), optional(:size) => 6226682, optional(:state) => binary(), optional(:url) => binary()}), optional(:assets_url) => binary(), optional(:body) => binary(), optional(:created_at) => binary(), optional(:draft) => false, optional(:id) => 53503802, optional(:name) => binary(), optional(:node_id) => binary(), optional(:prerelease) => true, optional(:published_at) => binary(), optional(:release_url) => binary(), optional(:tag_name) => binary(), optional(:tarball_url) => binary(), optional(:target_commitish) => binary(), optional(:upload_url) => binary(), optional(:url) => binary(), optional(:zipball_url) => binary()}})}
`

There are two issues,
1. Impossible to read all in one line. The output should be formatted.
2. Some atoms need double quotes, such as `:1.13.0` they should be represented as `:"1.13.0"`

if that is fixed, the report should look like this.

```elixir
%{
  required(:latest) => binary(),
  required(:releases) =>
    nonempty_list(
      {:"1.13.0",
       %{
         optional(:assets) =>
           nonempty_list(
             %{
               optional(:content_type) => binary(),
               optional(:created_at) => binary(),
               optional(:download_url) => binary(),
               optional(:id) => 50_943_932,
               optional(:label) => nil,
               optional(:name) => binary(),
               optional(:node_id) => binary(),
               optional(:size) => 2_023_670,
               optional(:state) => binary(),
               optional(:url) => binary()
             }
             | %{
                 optional(:content_type) => binary(),
                 optional(:created_at) => binary(),
                 optional(:download_url) => binary(),
                 optional(:id) => 50_943_934,
                 optional(:label) => nil,
                 optional(:name) => binary(),
                 optional(:node_id) => binary(),
                 optional(:size) => 6_230_020,
                 optional(:state) => binary(),
                 optional(:url) => binary()
               }
           ),
         optional(:assets_url) => binary(),
         optional(:body) => binary(),
         optional(:created_at) => binary(),
         optional(:draft) => false,
         optional(:id) => 54_599_470,
         optional(:name) => binary(),
         optional(:node_id) => binary(),
         optional(:prerelease) => false,
         optional(:published_at) => binary(),
         optional(:release_url) => binary(),
         optional(:tag_name) => binary(),
         optional(:tarball_url) => binary(),
         optional(:target_commitish) => binary(),
         optional(:upload_url) => binary(),
         optional(:url) => binary(),
         optional(:zipball_url) => binary()
       }}
      | {:"1.13.0-rc.0",
         %{
           optional(:assets) =>
             nonempty_list(
               %{
                 optional(:content_type) => binary(),
                 optional(:created_at) => binary(),
                 optional(:download_url) => binary(),
                 optional(:id) => 48_381_634,
                 optional(:label) => nil,
                 optional(:name) => binary(),
                 optional(:node_id) => binary(),
                 optional(:size) => 2_020_048,
                 optional(:state) => binary(),
                 optional(:url) => binary()
               }
               | %{
                   optional(:content_type) => binary(),
                   optional(:created_at) => binary(),
                   optional(:download_url) => binary(),
                   optional(:id) => 48_381_646,
                   optional(:label) => nil,
                   optional(:name) => binary(),
                   optional(:node_id) => binary(),
                   optional(:size) => 6_223_789,
                   optional(:state) => binary(),
                   optional(:url) => binary()
                 }
             ),
           optional(:assets_url) => binary(),
           optional(:body) => binary(),
           optional(:created_at) => binary(),
           optional(:draft) => false,
           optional(:id) => 52_477_144,
           optional(:name) => binary(),
           optional(:node_id) => binary(),
           optional(:prerelease) => true,
           optional(:published_at) => binary(),
           optional(:release_url) => binary(),
           optional(:tag_name) => binary(),
           optional(:tarball_url) => binary(),
           optional(:target_commitish) => binary(),
           optional(:upload_url) => binary(),
           optional(:url) => binary(),
           optional(:zipball_url) => binary()
         }}
      | {:"1.13.0-rc.1",
         %{
           optional(:assets) =>
             nonempty_list(
               %{
                 optional(:content_type) => binary(),
                 optional(:created_at) => binary(),
                 optional(:download_url) => binary(),
                 optional(:id) => 49_611_887,
                 optional(:label) => nil,
                 optional(:name) => binary(),
                 optional(:node_id) => binary(),
                 optional(:size) => 2_016_743,
                 optional(:state) => binary(),
                 optional(:url) => binary()
               }
               | %{
                   optional(:content_type) => binary(),
                   optional(:created_at) => binary(),
                   optional(:download_url) => binary(),
                   optional(:id) => 49_611_889,
                   optional(:label) => nil,
                   optional(:name) => binary(),
                   optional(:node_id) => binary(),
                   optional(:size) => 6_226_682,
                   optional(:state) => binary(),
                   optional(:url) => binary()
                 }
             ),
           optional(:assets_url) => binary(),
           optional(:body) => binary(),
           optional(:created_at) => binary(),
           optional(:draft) => false,
           optional(:id) => 53_503_802,
           optional(:name) => binary(),
           optional(:node_id) => binary(),
           optional(:prerelease) => true,
           optional(:published_at) => binary(),
           optional(:release_url) => binary(),
           optional(:tag_name) => binary(),
           optional(:tarball_url) => binary(),
           optional(:target_commitish) => binary(),
           optional(:upload_url) => binary(),
           optional(:url) => binary(),
           optional(:zipball_url) => binary()
         }}
    )
}
```

**Comment by eksperimental** [comment]
Number 3. Where is the error? 
I know it may not be the scope of Gradient but Gradualizer's .. but  I cannot figure out what is wrong. Where is the offending part specifically.

**Comment by erszcz** [comment]
Hi @eksperimental!

Thanks for reporting this. Indeed, both these cases are bugs - i.e. printing atoms and invalid inner map type inference. Thanks for spotting that!

I managed to make the warning in your code go away with [this change](https://github.com/erszcz/beam_langs_meta_data/commit/0cf357fef469fc9441dc94543625e44e437a5e36), but it obviously means the types guarantee way less about the returned values - and that's not what we want.

As you can see above, new tickets are already created in the appropriate places.

Thanks again!

**Comment by eksperimental** [comment]
> Thanks for reporting this. Indeed, both these cases are bugs - i.e. printing atoms and invalid inner map type inference. Thanks for spotting that!

No worries.
Thank for the fixes but it will not work for me. For the time being I will be using Dialyzer but as the bugs are ironed out (I am subscribed to the issues) I will be notified and I will add gradient to the test flow once these ones are fixed.

I will report more issues as I come across them.

**Comment by eksperimental** [comment]
I think both issues are fixed, please close if you agree with it.

**Comment by eksperimental** [comment]
There was a number 3, https://github.com/esl/gradient/issues/30#issuecomment-1030927630 
which is not fixed yet

**Comment by erszcz** [comment]
I'm closing this one as we either have more specific issues (#64, josefs/Gradualizer#378) or the issues are already fixed (#34, #43).

**Comment by eksperimental** [comment]
Thank you @erszcz 
I will be opening more issues, as I have added Gradient to my regular stack for new Elixir projects.
